### PR TITLE
feat: Add many-to-many relationships homework

### PR DESCRIPTION
**Pull request by SobollEva**
Add many-to-many relationships homework

### CODE REVIEW

**Comment by SobollEva** [comment]
image            
image        
image

File: src/hw-4/models/group.model.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            defaultValue: DataTypes.UUIDV4,
+            primaryKey: true,
+            allowNull: false
+        },
+        name: {
+            type: DataTypes.TEXT,
+            allowNull: false
+        },
+        permission: {
+            type: DataTypes.ARRAY(DataTypes.TEXT),
```
Comment (pavel-rudzenia) [inline src/hw-4/models/group.model.ts:16]: `permissions`, also it has to be an array of enum values (written down in the hometask).

File: src/hw-4/models/group.model.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            allowNull: false
+        }
+    }, {
+        timestamps: false,
+        tableName: 'group'
```
Comment (pavel-rudzenia) [inline src/hw-4/models/group.model.ts:21]: `Groups` according to the hometask description. I think it's not crucial here, but if there was several apps working with the table or several people working on the different layers such contract would be important.

File: src/hw-4/models/group.model.ts (hunk 1 above)
Comment (pavel-rudzenia) [reply src/hw-4/models/group.model.ts:16]: nevermind, it's going to be array of text values here. I think you can use [this link](https://sequelize.org/api/v6/class/src/data-types.js~enum) to define enum type, but seems it'll be supported only in postgre. But then you'll have ORM level validation for this field.

File: src/hw-4/models/userGroup.model.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { DataTypes, Sequelize } from 'sequelize';
+import dbConnection from '../services/db.service';
+
+import GroupModel from './group.model';
+import UserModel from './user.model';
+
+const UserGroupModel = (sequelize: Sequelize) => {
```
Comment (pavelrudenya) [inline src/hw-4/models/userGroup.model.ts:7]: be honest I don't understand why you need this UserGroup model when ORM does this linking between two models in similar way under the hood.

**Pull request merged**